src/main/java/org/jabref/gui/customizefields/CustomizeGeneralFieldsDialogViewModel.java

```diff
@@ -57,12 +58,13 @@ public void saveFields() {
                 return;
             }
 
-            String testString = BibtexKeyGenerator.cleanKey(parts[1], preferences.getEnforceLegalKeys());
+            String testString = BibtexKeyGenerator.cleanKey(parts[1], Globals.prefs.get(JabRefPreferences.DEFAULT_UNWANTED_BIBTEX_KEY_CHARACTERS));
```

Comment:
I think the second argument can be replaced by `preferences.getUnwantedCharacters()`.